### ISSUE TITLE
[java] UnnecessaryFullyQualifiedName false positive with similar package name

### ISSUE THREAD

**Issue by davidburstrom** [pmd/pmd#2299]
**Affects PMD Version:** 
6.20.0, at least.

**Rule:**
[UnnecessaryFullyQualifiedName](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#unnecessaryfullyqualifiedname)

**Description:**
If a FQN starts (but continues with further identifier symbols) with the current package name, there's a false positive `Unnecessary use of fully qualified name 'a.b.car.Car' due to existing same package import 'a.b.c.*'

**Code Sample demonstrating the issue:**

```java
package a.b.c;

class A {
  a.b.car.Car c;
}
```

**Running PMD through:**
CLI/Gradle

**Comment by adangel** [comment]
Fixed in PMD 7
